libsolidity/interface/CompilerStack.cpp

```diff
@@ -927,6 +926,17 @@ map<string, unsigned> CompilerStack::sourceIndices() const
 	return indices;
 }
 
+map<unsigned, shared_ptr<langutil::CharStream>> CompilerStack::indicesToCharStreams() const
+{
+	map<unsigned, shared_ptr<langutil::CharStream>> result;
+	unsigned index = 0;
+	for (auto const& s: m_sources)
+		result[index++] = s.second.scanner->charStream();
+	// NB: CompilerContext::yulUtilityFileName() does not have a source,
```

Comment:
After some more steps into completing this feature, we won't have the source of the others either. So maybe just add it with an empty shared pointer?